### PR TITLE
feat: add fullWsTransport support

### LINKED ISSUE #766: Queries and Mutations via WebSocket
I'm adding support for this feature in `mercurius`: https://github.com/mercurius-js/mercurius/pull/705

Once it lands I think we should add the same support in `graphql-hooks`.

The API will not be changed, in fact this one should just work:

```jsx
const client = createClient({
  url: "ws://localhost:4000/graphql",
});
```

Now it doesn't, the client throws `'GraphQLClient: config.fetch must be a function'`.

After that, the hooks will be used as usual, no changes.

At first it seems easy to implement but I'm not sure if it would clash with some existing features.

### PR DESCRIPTION
**Pull request by nuragic**
Add support for sending queries and mutations via WS, as proposed in [this issue](https://github.com/nearform/graphql-hooks/issues/766).

Tests are passing and I haven't touched them, so I think it's a good sign because this is not intended to be a breaking change. 😄 

### Next steps
- [x] validate proposal
- [x] add tests
- [x] add an example app on `examples`
- [x] update docs

Close #766.

### CODE REVIEW

**Comment by nuragic** [comment]
This branch can be tested with this client: https://github.com/nuragic/graphql-react-app/blob/graphql-hooks/

It involves some `npm link`_ing_ but it's fine 🔗 📦 🙃 

The only missing thing on that demo app is the manual cache update, but I opened another issue for that: #768

**Comment by simoneb** [comment]
Overall looks good so far, what's missing in this PR?

**Comment by simoneb** [comment]
Resolve the todo and write documentation, yes. Shall we have an example showing how to use this feature, too?

**Comment by nuragic** [comment]
Ok, I'll also add an example.

**Review by krisgrint — COMMENTED** [review]
Hi @nuragic just two tiny things with regard to the example app

**Comment by nuragic** [comment]
@krisgrint Thanks for the review! Just fixed those right now.

**Review by krisgrint — APPROVED** [review]
LGTM!

**Review by hails — APPROVED** [review]

File: packages/graphql-hooks/src/GraphQLClient.js [hunk 1]
```diff
@@ -189,6 +202,18 @@ class GraphQLClient {
   }
 
   request(operation, options = {}) {
+    if (this.fullWsTransport) {
+      return this.requestViaWS(operation)
+    } else {
```
Comment (simoneb) [inline packages/graphql-hooks/src/GraphQLClient.js:207]: redundant else

File: packages/graphql-hooks/src/GraphQLClient.js [hunk 2]
```diff
@@ -189,6 +202,18 @@ class GraphQLClient {
...
+      if (this.url) {
+        return this.requestViaHttp(operation, options)
+      } else {
```
Comment (simoneb) [inline packages/graphql-hooks/src/GraphQLClient.js:210]: redundant else

**Review by simoneb — CHANGES_REQUESTED** [review]
@hails a couple of comments, can you please address them?

**Review by simoneb — APPROVED** [review]

**Pull request merged**